pyxrf/model/draw_image_rgb.py

```diff
@@ -0,0 +1,353 @@
+            logger.info('Use scaler data to normalize,'
+                        'and the shape of scaler data is {}'.format(self.scaler_data.shape))
+        self.show_image()
+
+    @observe('scale_opt')
+    def _update_scale(self, change):
+        if change['type'] != 'create':
+            self.show_image()
+
+    def set_stat_for_all(self, bool_val=False):
+        """
+        Set plotting status for all the 2D images.
+        """
+        self.stat_dict.clear()
+        self.stat_dict = {k: bool_val for k in self.items_in_selected_group}
+
+    def preprocess_data(self):
+        """
+        Normalize data or prepare for linear/log plot.
+        """
+
+        selected_data = []
+        selected_name = []
+
+        stat_temp = self.get_activated_num()
+        stat_temp = OrderedDict(sorted(six.iteritems(stat_temp), key=lambda x: x[0]))
+
+        low_lim = 1e-4  # define the low limit for log image
+        ic_norm = 10000  # multiply by this value for ic normalization
+        plot_interp = 'Nearest'
+        name_not_scalable = ['r_squared']  # do not apply scaler norm on those data
+
+        if self.scaler_data is not None:
+            if np.max(self.scaler_data) < 1.0:  # use current as ic, such as SRX
+                ic_norm = 1.0
+            if len(self.scaler_data[self.scaler_data == 0]) > 0:
+                logger.warning('scaler data has zero values at {}'.format(np.where(self.scaler_data == 0)))
+                self.scaler_data[self.scaler_data == 0] = np.mean(self.scaler_data)
+                logger.warning('Use mean value {} instead for those points'.format(np.mean(self.scaler_data)))
+
+        if self.scale_opt == 'Linear':
+            for i, (k, v) in enumerate(six.iteritems(stat_temp)):
+
+                if self.scaler_data is not None:
+                    if k in name_not_scalable:
+                        data_dict = self.dict_to_plot[k]
+                    else:
+                        data_dict = self.dict_to_plot[k]/self.scaler_data*ic_norm
+
+                else:
+                    data_dict = self.dict_to_plot[k]
+
+                selected_data.append(data_dict)
+                selected_name.append(k) #self.file_name+'_'+str(k)
+
+        else:
+            for i, (k, v) in enumerate(six.iteritems(stat_temp)):
+
+                if self.scaler_data is not None:
+                    if k in name_not_scalable:
+                        data_dict = self.dict_to_plot[k]
+                    else:
+                        data_dict = self.dict_to_plot[k]/self.scaler_data*ic_norm
+
+                else:
+                    data_dict = np.log(self.dict_to_plot[k])
+
+                #maxz = np.max(data_dict)
+                selected_data.append(data_dict)
+                selected_name.append(k)
+
+        return selected_data, selected_name
+
+    def show_image(self):
+
+        selected_data, selected_name = self.preprocess_data()
+        selected_data = np.array(selected_data)
```

Comment:
make this `asarray` not `array` so it will only cast to an array if it needs to (http://stackoverflow.com/questions/14415741/numpy-array-vs-asarray)